src/order/bounded_lattice.lean

```diff
@@ -565,6 +565,14 @@ by rw [← sup_eq_max, lattice_eq_DLO]
 theorem inf_eq_min [linear_order α] (x y : with_bot α) : x ⊓ y = min x y :=
 by rw [← inf_eq_min, lattice_eq_DLO]
 
+@[simp, norm_cast, nolint simp_nf] -- the with_top lemmas leak through when applying simp
```

Comment:
Doesn't that mean that this simp lemma isn't actually used by `simp`, as it finds the other lemma first?

Reply:
I had the same issue in #6856: https://github.com/leanprover-community/mathlib/pull/6845/files#diff-4880929e879aabbb6e5cecbecc734d3c1b1aaf1bace3754ea2c5ffd4e8076d86, and so far have just planned to remove the simp lemmas where `simp_nf` was failing.

Reply:
@fpvandoorn, does something go wrong if we don't mark this with `@[simp]`?

Reply:
No, that should be fine. I removed the simp and changed the comment.

Reply:
Should I also remove the norm_cast?

Reply:
Are the `with_top` lemmas in question also marked `norm_cast`? If so, then these lemmas might not ever be triggered either, right?

Reply:
They are. I don't know if `norm_cast` tests using the same reducibility settings as `simp`, so I'm not sure about your last question.

Reply:
I just tested it, without the attribute `norm_cast` fails, so we should keep the `norm_cast` attribute.
```lean
import order.bounded_lattice

local attribute [-norm_cast] with_bot.coe_max
example {α} [linear_order α] (x y : α) : ((max x y : α) : with_bot α) = max x y :=
by show_term { norm_cast } -- fails
```